caveclient/skeletonservice.py

```diff
@@ -0,0 +1,116 @@
+
+from .auth import AuthClient
+from .base import (
+    ClientBase,
+    _api_endpoints,
+    handle_response,
+)
+from .endpoints import skeletonservice_api_versions, skeletonservice_common
+from typing import Optional
+
+SERVER_KEY = "skeleton_server_address"
+
+
+'''
+Usage
+'''
+
+class SkeletonClient(ClientBase):
+    def __init__(
+        self,
+        server_address,
+        auth_client=None,
+        api_version="latest",
+        verify=True,
+        max_retries=None,
+        pool_maxsize=None,
+        pool_block=None,
+        over_client=None,
+    ):
+        if auth_client is None:
+            auth_client = AuthClient()
+
+        auth_header = auth_client.request_header
+        endpoints, api_version = _api_endpoints(
+            api_version,
+            SERVER_KEY,
+            server_address,
+            skeletonservice_common,
+            skeletonservice_api_versions,
+            auth_header,
+            fallback_version=1,
+            verify=verify,
+        )
+
+        super(SkeletonClient, self).__init__(
+            server_address,
+            auth_header,
+            api_version,
+            endpoints,
+            SERVER_KEY,
+            max_retries=max_retries,
+            pool_maxsize=pool_maxsize,
+            pool_block=pool_block,
+            over_client=over_client,
+        )
+    
+    def get_skeleton_info(
+        self,
+        datastack_name,
+    ):
+        """Gets basic skeleton information for a datastack
+
+        Parameters
+        ----------
+        datastack_name : str
+            The name of the datastack to check
+
+        Returns
+        -------
+        bool
+            JSON description
+        """
+        endpoint_mapping = self.default_url_mapping
+        endpoint_mapping["datastack_name"] = datastack_name
+
+        url = self._endpoints["skeleton_info"].format_map(endpoint_mapping)
+
+        response = self.session.get(url)
+        return handle_response(response, False)
+    
+    def get_skeleton(
+        self,
+        root_id: int,
+        datastack_name:   Optional[str] =None,
+        skeleton_version: Optional[int] =0,
```

Comment:
Why not use "skeleton_version = None" for latest? This would be implied by the Optional argument, which would imply the `skeleton_version` parameter takes `None`. Also, I think I'm surprised that versions are necessarily integers — in a world where datastacks have individual skeletonization parameters and methods, I would think that a short name would often be convenient for descriptions.

Reply:
How do you pass a none into a REST URL parameter? It is represented as a number, an integer, for this reason.